src/backend/distributed/planner/multi_planner.c

```diff
@@ -127,6 +127,18 @@ multi_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 			result = CreateDistributedPlan(result, originalQuery, parse,
 										   boundParams, restrictionContext);
 		}
+		else
+		{
+			needsDistributedPlanning = NeedsDistributedPlanning(parse);
```

Comment:
I think we can do this check after PG_TRY()/PG_CATCH() blocks just before returning. 

We can also use just one if block such as;

```C
	if (!needsDistributedPlanning && NeedsDistributedPlanning(parse))
	{
		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
						errmsg("cannot perform distributed planning on this "
							   "query because parametrized queries for SQL "
							   "functions referencing are not supported."),
						errhint("Consider using PLPGSQL functions instead.")));
	}
```

I think this is nicer because it shows in one line that it was not distributed but now it returns distributed so something is wrong :)